permissions/resource_rule_set.go

```diff
@@ -0,0 +1,139 @@
+package permissions
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceRuleSet() *schema.Resource {
+	s := common.StructToSchema(
+		iam.RuleSetUpdateRequest{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["etag"].Required = false
+			m["etag"].Computed = true
+			m["grant_rules"].Type = schema.TypeSet
+			common.MustSchemaPath(m, "grant_rules", "principals").Type = schema.TypeSet
```

Comment:
I'm not sure that we should use `set` type instead of the `list` at least for the `grant_rules` blocks - we had problems in DLT resource when set was used for complex objects, then it caused nor necessary diffs, etc.

Reply:
I did not notice any issues with my local testing. This is similar to what is done by UC - https://github.com/databricks/terraform-provider-databricks/blob/master/catalog/resource_grants.go#L39

Reply:
See #1401 and PRs #1416 & #1508 about effect of using `TypeSet` with complex structures.

Reply:
Thanks @alexott for the review. Going through the PRs, was it investigated why empty structs were generated? 

> But it should be investigated why we generated empty structs when using reflection

Reply:
@alexott should I change the schema of grant_rules to list? The backend will handle duplicate entries of the same role so it should be fine from that perspective.

Reply:
This is really a good question - and we need to check if backend may return the entries in the different order? Like, if I submit

```
["z1", "a1", "x1", "b1"]
```

may it return things in the other order?

```
["a1", "b1", "x1", "z1"]
```

if yes, then it may cause constant configuration drift.

Reply:
the backend will merge the entires..

```
{
  "role": "servicePrincipal/user",
  "principals": ["users/gautham@databricks.com"]
}

{
  "role": "servicePrincipal/user",
  "principals": ["groups/data_scientists"]
}
```

will become
```
{
  "role": "servicePrincipal/user",
  "principals": ["users/gautham@databricks.com", "groups/data_scientists"]
}
```

Reply:
Actually, let me verify this..

Reply:
If it is definite (and not just this example instance done above) that backend returns in specified order, then what might be the issue with using this assumption?

Reply:
@gauthamsunjay I think this is the only unresolved issue right? Rest looks lgtm to me.

Reply:
Is it a blocker to have the grant_rules as a set? If the backend behavior changes there will be constant drift which we don't want as well. 

cc @alexott

Reply:
let leave it as set - if you didn't find any problems in testing, then it could work. I simply had bad experience before. But we can change type if necessary